### DISCUSSION TITLE
Using Directus with Supabase

### DISCUSSION THREAD

**Discussion by cspecter** [directus/directus#9057] [Feature Requests]
We are using Directus as our CMS for a Supabase hosted database. We would like to be able to use Directus to manage file and file associations in our tables in Supabase. My understanding is that Supabase uses S3 buckets for their storage, but that they don't give you direct access to them. You have to go through their storage API. In Postgres, Supabase has a `storage.files` table that holds the info to the file. We duplicated this table to `public.files` so we could link to files from our public tables.

We would like to get two things working in Directus, if possible. First, it would be great if we could get image file previews when editing our collections. We would probably need to use the Supabase storage API for this. Is there a type that will let us run custom code for this?

Second, it would be great to be able to utilize the Supabase storage API so the app manager could upload files via Directus. Is there a way to make use of the built-in storage adapters for this? Seems like we would need to use the Supabase storage API though.

**Comment by rijkvanzanten** [comment]
Heya! I'm not too familiar with the internal workings of Supabase, but it sounds like the storage solution they offer isn't directly compatible with Directus, as it sounds like it's a proprietary storage API which isn't directly s3 compatible 🤔 I don't think there's a way currently to hack your way through hooks to override every upload/download method. The best approach would be to look at an existing storage adapter (like https://github.com/directus/directus/tree/main/packages/drive-s3) and open a PR for a dedicated drive-supabase package that uses the storage APIs offered by supabase 👍🏻

**Reply by cspecter** [reply]
Good suggestions. I'll take a look and see if there is an easy way to add that kind of functionality and make a feature request.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
